web/packages/teleterm/src/mainProcess/protocolHandler.ts

```diff
@@ -74,7 +88,55 @@ function interceptFileProtocol() {
   });
 }
 
+/**
+ * Registers the 'app-file://' protocol handler.
+ * Serves application files from the build directory and adds
+ * cross-origin headers to responses, enabling features that
+ * require cross-origin isolation.
+ */
+function handleAppFileProtocol(): void {
+  const appPath = app.getAppPath();
+
+  protocol.handle(APP_FILE_SCHEMA, async request => {
+    const filePath = decodeURIComponent(new URL(request.url).pathname).replace(
+      // Remove the leading slash.
+      /^\/+/,
+      ''
+    );
+    const target = path.join(appPath, filePath);
+
+    // Use net.fetch to serve local files.
+    // It automatically determines and sets the correct Content-Type (MIME) header,
+    // unlike fs.readFile.
+    const response = await net.fetch(pathToFileURL(target).toString());
+    // To use features like SharedArrayBuffer, the document must be in a secure context
+    // and cross-origin isolated.
+    response.headers.set('Cross-Origin-Opener-Policy', 'same-origin');
+    response.headers.set('Cross-Origin-Embedder-Policy', 'require-corp');
+    return response;
+  });
+}
+
+/**
+ * Registers the 'app-file://' protocol schema.
+ * It's used to serve application files.
+ */
+export function registerAppFileProtocol(): void {
+  protocol.registerSchemesAsPrivileged([
+    {
+      scheme: APP_FILE_SCHEMA,
+      privileges: {
+        standard: true,
+        secure: true,
+        supportFetchAPI: true,
+        codeCache: true,
```

Comment:
Do we need to enable `codeCache` if we ever only use a single window?

From what I understand, this would help if we were running the same script many times. But we load each of the big ones (renderer, shared, main) just once.